core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java

```diff
@@ -214,4 +235,412 @@ public static String getMetadataObjectFullName(String type, long metadataObjectI
+    List<ColumnPO> columnPOs =
+        SessionUtils.getWithoutCommit(
+            TableColumnMapper.class, mapper -> mapper.listColumnPOsByColumnIds(ids));
+
+    if (columnPOs == null || columnPOs.isEmpty()) {
+      return new HashMap<>();
+    }
+
+    List<Long> catalogIds =
+        columnPOs.stream().map(ColumnPO::getCatalogId).collect(Collectors.toList());
+    List<Long> schemaIds =
+        columnPOs.stream().map(ColumnPO::getSchemaId).collect(Collectors.toList());
+    List<Long> tableIds = columnPOs.stream().map(ColumnPO::getTableId).collect(Collectors.toList());
+
+    Map<Long, String> catalogIdAndNameMap = getCatalogIdAndNameMap(catalogIds);
+    Map<Long, String> schemaIdAndNameMap = getSchemaIdAndNameMap(schemaIds);
+    Map<Long, String> tableIdAndNameMap = getTableIdAndNameMap(tableIds);
+
+    HashMap<Long, String> columnIdAndNameMap = new HashMap<>();
+
+    columnPOs.forEach(
+        columnPO -> {
+          // since the catalog or schema or table can be deleted, we need to check the
+          // null value, and when catalog or schema or table is deleted, we will set
+          // fullName of columnPO to null
+          String catalogName = catalogIdAndNameMap.getOrDefault(columnPO.getCatalogId(), null);
+          if (catalogName == null) {
+            LOG.warn("The catalog of column {} may be deleted", columnPO.getColumnId());
+            columnIdAndNameMap.put(columnPO.getColumnId(), null);
+            return;
+          }
+
+          String schemaName = schemaIdAndNameMap.getOrDefault(columnPO.getSchemaId(), null);
+          if (schemaName == null) {
+            LOG.warn("The schema of column {} may be deleted", columnPO.getColumnId());
+            columnIdAndNameMap.put(columnPO.getColumnId(), null);
+            return;
+          }
+
+          String tableName = tableIdAndNameMap.getOrDefault(columnPO.getTableId(), null);
+          if (tableName == null) {
+            LOG.warn("The table of column {} may be deleted", columnPO.getColumnId());
+            columnIdAndNameMap.put(columnPO.getColumnId(), null);
+            return;
+          }
+
+          String fullName =
+              DOT_JOINER.join(catalogName, schemaName, tableName, columnPO.getColumnName());
+          columnIdAndNameMap.put(columnPO.getColumnId(), fullName);
+        });
+
+    return columnIdAndNameMap;
+  }
+
+  /**
+   * Retrieves a map of Catalog object IDs to their full names.
+   *
+   * @param ids A list of Catalog object IDs to fetch names for.
+   * @return A Map where the key is the Catalog ID and the value is the Catalog full name. The map
+   *     may contain null values for the names if its parent object is deleted. Returns an empty map
+   *     if no Catalog objects are found for the given IDs. {@code @example} value of catalog full
+   *     name: "catalog1"
+   */
+  public static Map<Long, String> getCatalogObjectFullNames(List<Long> ids) {
```

Comment:
Is it possible to merge `getCatalogObjectFullNames` and `getSchemaIdAndNameMap`?

Reply:
No we can't, because methods like `getTableObjectFullNames` also use `getSchemaIdAndNameMap`.